vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -981,7 +993,8 @@ bool AdsImpl::IsAllowedToShowAds() {
       << does_history_respect_ads_per_day_limit;
 
   return does_history_respect_minimum_wait_time &&
-      does_history_respect_ads_per_day_limit;
+      does_history_respect_ads_per_day_limit && 
+      notifications_->Count() < ads_client_->GetAdsPerSameTime();
```

Comment:
This use case is only for mobile devices, and should not affect the desktop browser.

Please move the business logic to the following code block:

```
  bool does_history_respect_minimum_wait_time;
  if (!IsMobile()) {
    does_history_respect_minimum_wait_time =
        DoesHistoryRespectMinimumWaitTimeToServeAds();
  } else {
    does_history_respect_minimum_wait_time = true;
  }
```

Reply:
Reading the requirements this should be a FIFO queue, so this is not the correct place to add this logic. So I would recommend adding a function along the lines of which is called from `AdsImpl::ServeAd`:

```
auto notifications_count = notifications_->Count();
If (notifications_count >= kMaximumAdNotifications) {
   notifications_->RemoveAt(notifications_count - 1);
}
```

Reply:
@tmancey 
Unfortunately, posted notifications (in notifications.cc) are stored in the std::map which is sorted by id. So to implement a FIFO we need a queue container or store a post timestamp for each notification (which we discussed yesterday).

Reply:
@gdregalo As discussed agreed we will change from map to queue as only currently used for Android so a good time to make the change

Reply:
done